### ISSUE TITLE
Suggestion for model.train - returning best model parameters

### ISSUE THREAD

**Issue by rowlesmr** [GAMES-UChile/mogptk#57]
In a few of my tests, the loss (and error) both increase and decrease with iterations. Sometimes, I have a loss which is smaller somewhere in the middle iterations than at the last iteration.

Is it possible to either always return the model parameters corresponding to the minimum loss (or error), or add a flag which enables this behaviour?

**Comment by tdewolff** [comment]
Thanks for the idea, I will take a look if this can be implemented soon. The only objection would be that your model loss should be decreasing in the long run, and the only reason for jitter that increases the loss is the optimizer trying to find a better set of parameters.

**Comment by rowlesmr** [comment]
Yes, but if you happen to stop iterating in the middle of a jitter, then you get stuck with sub optimal parameters.

**Comment by tdewolff** [comment]
I don't think we'll implement this, it would boil down to cherry-picking the best parameters, while you should perhaps continue training to obtain those better parameters, or tweak the learning rate.
